File: src/simulation/FitnessResult.java
package simulation;

import neat.Genotype;

/**
 * Created by colander on 2/1/17.
 * Class used as a sortable container for the results of fitness measurements.
 */
public class FitnessResult implements Comparable<FitnessResult> {
    int id;
    public double result;
    public Genotype genotype;

    public FitnessResult(double result, Genotype genotype) {
        this.result = result;
        this.genotype = genotype;
    }

    public FitnessResult(double result, Genotype genotype, int id) {
        this.result = result;
        this.genotype = genotype;
        this.id = id;
    }

    public int compareTo(FitnessResult o) {
        return this.result != o.result ? Double.compare(this.result, o.result) : Integer.compare(this.id, o.id);
    }
}

File: src/main/java/com/janboucek/crawler/simulation/FitnessResult.kt
package com.janboucek.crawler.simulation

import com.janboucek.crawler.neat.Genotype

/**
 * Created by colander on 2/1/17.
 * Class used as a sortable container for the results of fitness measurements.
 */
data class FitnessResult(var result: Double, var genotype: Genotype, private var id: Int = 0) : Comparable<FitnessResult> {

    override fun compareTo(other: FitnessResult): Int {
        return if (result != other.result) result.compareTo(other.result) else id.compareTo(other.id)
    }
}